### DISCUSSION TITLE
How to config a single nats-server with jetstream in most resource-effective way

### DISCUSSION THREAD

**Discussion by tpihl** [nats-io/nats-server#4227] [Q&A] [answered]
Hi,

I am making a PoC using nats-server (and go-based nats client) without network access on a battery-powered SBC.

However, the nats-server seems to keep itself rather busy (relative term but significant for battery time) and I hope for help in calming it down when the rest of the world is carm.

Changing PING interval was easy but it seems to have a very small effect (most activity seems to be related to
$SYS.>
$JSC.>

Are these really necessary in a single instance nats-server with a single client in the same machine. If they are, can we reduce the frequency of gathering/sending that info when there is no other activity?

I havent tried to run in MQTT mode even though you probably guess that it's MQTT server this PoC competes with. My argument is borrowed from how i remember a  nats.io recommendation, something like "use real nats/jetstream if possible, and that it should prove as effective in a comparable scenario with MQTT"

Any ideas/suggestions?

/T

**Comment by Jarema** [comment]
I am moving this to `nats-server`, as it's not a client-related discussion.

**Comment by derekcollison** [comment]
Let me take a look, if the $JSC subjects are specifically `$JSC.ARU.>` agree there is no need for those per se in a single server setup unless folks are monitoring those themselves.

**Reply by derekcollison** [reply]
I have those suppressed in not in JetStream cluster mode.

**Comment by derekcollison** [comment]
Any specific `$SYS` subjects you want to call out.

**Reply by derekcollison** [reply]
I will make statsz updates only go out if there is external interest.

**Comment by derekcollison** [comment, marked as answer]
Made the improvements and they will be in 2.9.18 this week.

**Reply by tpihl** [reply]
Will check them out, thank you!
